clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,19 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+
+      // if the Action is of OffloadAction type, use the HostAction of the
+      // specfic Offload Action to set LastArg
+      if (auto *IA = dyn_cast<OffloadAction>(UnbundlerInputs.back()))
+        LastArg =
+            &(dyn_cast<InputAction>(IA->getHostDependence())->getInputArg());
+      else if (auto *IA = dyn_cast<InputAction>(UnbundlerInputs.back()))
+        // else set the LastArg based on Last InputAction
+        LastArg = &(IA->getInputArg());
+      else
+        Diag(clang::diag::warn_invalid_action)
+            << UnbundlerInputs.back()->getClassName();
```

Comment:
From a user perspective, letting them know that we are encountering an unexpected internal action type is not desired.  They definitely wouldn't know what to do with that information.  We know that we are only expecting 2 types of actions (InputAction, OffloadAction) here, so I think an assert would be better.

Reply:
Done, changed as suggested.